packages/router-plugin/src/core/router-code-splitter-plugin.ts

```diff
@@ -16,11 +16,14 @@ function capitalizeFirst(str: string): string {
 }
 
 function fileIsInRoutesDirectory(filePath: string, routesDirectory: string) {
-  const routesDirectoryPath = isAbsolute(routesDirectory)
-    ? routesDirectory
-    : join(process.cwd(), routesDirectory)
-
-  return filePath.startsWith(routesDirectoryPath)
+  const routesDirectoryPath = normalize(
+    isAbsolute(routesDirectory)
+      ? routesDirectory
+      : join(process.cwd(), routesDirectory),
+  )
+  const path = normalize(filePath)
```

Comment:
only this `normalize` is then needed, the `routesDirectoryPath` should already be correct.

Can you please test this and remove if my assumption is correct?

Reply:
Yes. `path.resolve` do the `normalize` too.

I have updated the fork and rerun the CI, and the CI was pass.

The full logs here: https://github.com/greenhat616/reproduce-vite-code-splitting/actions/runs/10928570228/job/30340534961

Reply:
thanks, can you fix the type issues please? see the CI run here: https://cloud.nx.app/runs/VFxSYgAzpD/task/%40tanstack%2Frouter-plugin%3Atest%3Atypes